02_commands.html

```diff
@@ -1,206 +1,524 @@
+
+          <li><p>If <var>success</var> is false, or <var>success</var> is true and <var>parameters</var> is not an Object object, <a>return an error</a> with status <a>error code</a> <a>invalid argument</a> and jump back to step 1 in this overall algorithm.</p></li>
+        </ol>
+
+        <p>Otherwise, let <var>parameters</var> be null</p>
+      </li>
+
+      <li><p>Let <var>response data</var> be a <a>command response</a> object obtained by running running the steps defined by <var>command</var> with arguments <var>element id</var> and <var>parameters</var>.</p></li>
+
+      <li><p>If <var>response data</var> is an <a>error</a>, <a>return an error</a> with <a>error code</a> equal to <var>response data</var>'s <a>error code<a>.</p>
+
+        <p>Otherwise <a>send a response</a> with status 200 and <var>response data</var>'s data.</p></li>
+
+      <li><p>Jump to step 1</p></li>
+    </ol>
+
+    <p>When required to <dfn>send a response</dfn>, with arguments <var>status</var> and <var>data</var>, a remote end must run the following steps:</p>
+
+    <ol>
+      <li><p>Let <var>response</var> be a new <a href="https://fetch.spec.whatwg.org/#concept-response" class="externalRef">response</a>.</p></li>
+
+      <li><p>Set <var>response</var>'s <a href="https://fetch.spec.whatwg.org/#concept-response-status" class="externalRef">status</a> to <var>status</var>, and <a href="https://fetch.spec.whatwg.org/#concept-response-status-message" class="externalRef">status message</a> to the string corresponding to the description of <var>status</var> in the <a href="http://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml" class="externalRef">status code registry</a>.</p></li>
+      <li><p>If <var>data</var> is not null, let <var>response</var>'s <a href="https://fetch.spec.whatwg.org/#concept-response-body" class="externalRef">body</a> be the result of <a>serializing as JSON</a> with <var>data</var> as the argument.</p></li>
+
+      <li><p>Serialize <var>response</var> according to the rules in [[!RFC7230]] and write the resulting bytes to the <a>connection</a></p></li>
+
+    </ol>
+
+    <p>When required to <dfn>return an error</dfn>, with <var>error code</var>, run the following steps:</p>
+
+    <ol>
+      <li><p>Let <var>http status</var> and <var>name</a> be the <a>error response data</a> for <var>error code</var></p></li>
+      <li><p>Let <var>data</var> be a <a>new JSON Object</a>.</p>
+
+      <li><p>Let <var>message</var> be an implementation-defined string containing a human-readable description of the reason for the error.</p></li>
+
+        <!-- TODO: really need a better way of constructing JSON literals -->
+      <li><p><a>Set the properties</a> of <var>object</var> with values ("error", <var>name</var>), ("message", <var>message</var>).</p></li>
+
+      <li><a>Send a response</a> with status <var>status</var> and data <var>data</var></li>
```

Comment:
“Send a response with status and data as arguments” I think would be better.

Missing full stop.